Connection acquisition timed out for session mcp_session

Frequently encounters the connection acquisition timed out error, the query are just simple count queries...


```
2025-07-22 02:15:10.621 [INFO]     doris_mcp_server.main.DorisServer - MCP Request - Headers: {b'host': b'13.158.18.151:3000', b'accept-encoding': b'gzip, deflate', b'connection': b'keep-alive', b'user-agent': b'python-httpx/0.27.2', b'content-type': b'application/json', b'accept': b'application/json, text/event-stream', b'mcp-session-id': b'7ba3ec999e7d4c84bd2d249ccabfcb3c', b'content-length': b'166'}
2025-07-22 02:15:10.622 [INFO]     mcp.server.lowlevel.server - Processing request of type CallToolRequest
2025-07-22 02:15:10.624 [INFO]     doris_mcp_server.main.DorisServer - Handling tool call request: exec_query
2025-07-22 02:15:10.624 [INFO]     doris_mcp_server.utils.schema_extractor - Executing SQL query: SELECT COUNT(*) as row_count FROM internal.ssb.customer, DB: None, Catalog: None, MaxRows: 100, Timeout: 30
2025-07-22 02:15:20.625 [ERROR]    doris_mcp_server.utils.db - Connection acquisition timed out for session mcp_session
```

Is it caused by the connection pool? If so, is there a way to configure the connection pool?

### ENV

* doris-mcp-server 0.5.1
* doris 2.1.9
* OS: Ubuntu 22.04

--- a/doris_mcp_server/utils/query_executor.py
+++ b/doris_mcp_server/utils/query_executor.py
@@ -432,21 +432,17 @@ class DorisQueryExecutor:
         )
 
         # Execute query
-        connection = await self.connection_manager.get_connection(
-            query_request.session_id
-        )
-
         # Set timeout if specified
         if query_request.timeout:
             try:
                 result = await asyncio.wait_for(
-                    connection.execute(optimized_sql, query_request.parameters, auth_context),
+                    self.connection_manager.execute_query(query_request.session_id, optimized_sql, query_request.parameters, auth_context),
                     timeout=query_request.timeout
                 )
             except asyncio.TimeoutError:
                 raise Exception(f"Query timeout after {query_request.timeout} seconds")
         else:
-            result = await connection.execute(optimized_sql, query_request.parameters, auth_context)
+            result = await self.connection_manager.execute_query(query_request.session_id, optimized_sql, query_request.parameters, auth_context)
 
         return result
 

